carbon-design-system/carbon-addons-iot-react
# feat(table): add option to pin column

Closes #3636

**Summary**

- Add new prop to `options` to pin first or last column

**Change List (commits, features, bugs, etc)**

- Add new css classes to freeze column position
- Setup with different properties, such as row selection and nesting
- Create new story in Storybook
- Update public API
- Add new tests

**Acceptance Test (how to verify the PR)**

- _With row selection_
- Go to [Table / With pinned column](https://deploy-preview-3677--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--with-pinned-column)
- Resize window horizontally so that scrollbar appears
- Using horizontal scrollbar in table scroll to the right
- Verify that first column is pinned
- In the knobs area set `pinColumn: last`
- Verify that last column is pinned

</br>

- _With row nesting_
- Go to [Table / Playground](https://deploy-preview-3677--carbon-addons-iot-react.netlify.app/?path=/story/1-watson-iot-table--playground)
- In the `Nesting & expansion` set `hasRowExpansion: true` and `hasRowNesting: true`
- In the `Selection & actions` set `hasRowSelection: false`
- In the `Column configuration` set `pinColumn: first`
- Resize window horizontally so that scrollbar appears
- Using horizontal scrollbar in table scroll to the right
- Verify that first column is pinned
- _Latest update:_ If in `Data rendering` knob `wrapCellText: always` then overflow content will be hidden (For example, 5th row)

**Regression Test (how to make sure this PR doesn't break old functionality)**

- tests here



**Things to look for during review**
